compiler/rustc_data_structures/src/steal.rs

```diff
@@ -51,6 +51,15 @@ impl<T> Steal<T> {
         let value = value_ref.take();
         value.expect("attempt to steal from stolen value")
     }
+
+    /// Writers of rustc drivers often encounter stealing issues. This function makes it possible to
+    /// handle these errors gracefully.
+    ///
+    /// This should not be used within rustc as it leaks information not tracked
+    /// by the query system, breaking incremental compilation.
+    pub fn is_stolen(&self) -> bool {
```

Comment:
Maybe call it `untracked_is_stolen()`? We tend to prefix methods that don't properly go through the incr comp system with `untracked_`.

Reply:
Ah, I didn't know that. I was in the process of adding an internal lint to warn about uses of this function as was suggested ^^. Do you think the lint would be worth it?

Reply:
There are probably other methods which don't have the `untracked_` prefix. Either renaming those or adding a lint for them would be beneficial I think.